temporalio/worker/_workflow.py

```diff
@@ -193,6 +206,48 @@ async def _handle_activation(
     ) -> None:
         global LOG_PROTOS
 
+        def get_traceback_for_thread(
+            thread_id: int,
+        ) -> Tuple[Optional[TracebackType], bool]:
```

Comment:
I don't think this function should be nested inside `_handle_activation`. In general, we should nest functions as little as possible, e.g. one valid reason is if it's necessary to write a closure that has to mutate a variable in outer scope. But this one isn't closing over anything, and in fact doesn't even need to be a method -- it's just a pure function, right? I think that moving it out doesn't affect the frame logic but lmk if that's wrong!

Reply:
My vote is since it isn't closely associated with what this class is about, it should just be a top-level function rather than a static method.